planning/MartenV4-WastazComments.md

```diff
@@ -0,0 +1,174 @@
+# Projections
+
+## Building projections
+While its nice that Marten provides several options for actually constructing projections (`ITransform`, `AggregateStreamWith`, `ViewProjection`, 
+raw `IProjection`, etc...honestly Ive probably forgotten some way of doing it) in the end the ones that I end up using 99% of the time is
+either `ViewProjection` or implementing the raw `IProjection`. The main reason for this is that most of the other options are too limiting
+when building projections. 
+
+Working with `ViewProjection` is quite nice, however occasionally the number of overloads can make it hard because its absolutely not hard
+to make the compiler confused over which overload you are actually trying to use. The main thing that I cant do with `ViewProjection` which
+often is the thing that makes me drop down to `IProjection` is project an event to multiple documents, which is something that can be very useful
```

Comment:
Yes, that was also my impression, as I stated in the other comment it would be good to provide some simpler abstraction than just `IProjection`. ViewProjection (besides too many conflicting overloads) seems to just do the job.

`IProjection` interface tho is I think good as the abstraction for the general subscription model that someone may create its own subscription. (maybe it should be renamed to such?).

Reply:
@oskardudycz This is why I was pushing the Transformer class idea to bypass the overload hell you get from the `ViewProjection` approach.